### ISSUE TITLE
`git_branch_upstream` could not get the upstream branch in `config.worktree`

### ISSUE THREAD

**Issue by Whirlwind** [libgit2/libgit2#6650]
### Reproduction steps

```
$ git clone --filter=blob:none git@xxx:xx/xx.git
$ cd xx
$ git worktree add ../worktree
$ cd ../worktree
$ git config extensions.worktreeConfig true
$ git checkout new-branch
$ git push -u
```

Then, use the `git_branch_upstream` will could not get the `origin/new-branch`, it said `GIT_ENOTFOUND`.

### Expected behavior

`git_branch_upstream` can get the upstream branch in the `config.worktree`.

### Actual behavior

throw error.

### Version of libgit2 (release number or SHA1)

1.6.4

### Operating system(s) tested

macOS 14.1

**Comment by csware** [comment]
cf. https://github.com/libgit2/libgit2/issues/6044

**Comment by ethomson** [comment]
Should be fixed via #6756
